library/common/src/main/java/com/epmedu/animeal/common/di/AppModule.kt

```diff
@@ -0,0 +1,41 @@
+package com.epmedu.animeal.common.di
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.core.handlers.ReplaceFileCorruptionHandler
+import androidx.datastore.preferences.SharedPreferencesMigration
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.preferencesDataStoreFile
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import javax.inject.Singleton
+
+private const val DATA_STORE_PREFERENCES_KEY = "preferences"
+
+@Module
+@InstallIn(SingletonComponent::class)
+internal object AppModule {
+
+    @Singleton
+    @Provides
+    fun providesPreferencesDataStore(
+        @ApplicationContext context: Context
+    ): DataStore<Preferences> {
+        return PreferenceDataStoreFactory.create(
+            corruptionHandler = ReplaceFileCorruptionHandler(
+                produceNewData = { emptyPreferences() }
+            ),
+            migrations = listOf(SharedPreferencesMigration(context, DATA_STORE_PREFERENCES_KEY)),
```

Comment:
do we really need migrations? Previously we didn't have shared preferences

Reply:
the same for `produceFile`

Reply:
Migration removed. But `produceFile` is necessary